src/wireviz/wv_output.py

```diff
@@ -220,3 +218,35 @@ def generate_titlepage(yaml_data, extra_metadata, shared_bom, for_pdf=False):
         rendered={'index_table': index_table.render(options)},
         bom_render_options=bom_render_options,
     )
+
+
+def generate_xlsx_output(filename, inp, header=None):
+    print('Generating xlsx output')
+    workbook = xlsxwriter.Workbook(filename)
+    worksheet = workbook.add_worksheet()
+
+    populate_spreadsheet_header_row(worksheet, get_xlsx_row_ordered_dict(inp.entries[0]))
+    for row_idx, line_item in enumerate(inp.entries):
+        line_item = get_xlsx_row_ordered_dict(line_item)
+        for field in line_item:
+            worksheet.write(row_idx+1, list(line_item.keys()).index(field), line_item[field])
+
+        qty_to_buy_formula = f'= INDIRECT(Address({row_idx+1}, {get_excel_col_idx(line_item, "Qty")})) + INDIRECT(Address({row_idx+1}, {get_excel_col_idx(line_item, "Qty Extras Desired")})) - INDIRECT(Address({row_idx+1}, {get_excel_col_idx(line_item, "Qty Inventory")}))'#+1 because title row
+        worksheet.write_formula(row_idx+1, get_excel_col_idx(line_item, 'Qty to Buy'), qty_to_buy_formula)# how to eliminate assumption that qty to buy is last field?
+    workbook.close()
+
+def get_excel_col_idx(input_row, col_name):
+    return list(input_row.keys()).index(col_name) + 1
+
+def populate_spreadsheet_header_row(worksheet, line_item):
+    for idx, key in enumerate(line_item):
+        worksheet.write(0, idx, key)
```

Comment:
from generate_xlsx_output to here probably warrent its own representation.

This way you can implement more complex calls into methods.
Particularly, the qty_to_buy_formula.

Reply:
You mean make an xlsx generator object?